doc/reference/vm_live_migration_internals.md

```diff
@@ -0,0 +1,127 @@
+1. Snapshot sync for non-shared storage to ensure consistency between source and target.
+1. Transfer VM state to target.
+
+```{figure} /images/vm_live_migration_flowchart.svg
+Non-Shared Storage Migration
+```
+
+The state transitions during the process are shown below:
+
+```{figure} /images/vm_live_migration_fsa_diagram.svg
+Non-Shared Storage Migration State Transitions
+```
+
+### Intra-cluster member live migration (Ceph shared storage pool)
+
+For shared storage pools such as Ceph, disk data transfer is unnecessary. Instead, the process focuses on transferring the VM state through a dedicated migration socket:
+
+1. Validate cluster state and storage pool readiness.
+1. Notify the shared disks that they will be accessed from another system.
+1. Pause the guest OS on the source VM and transfer the live state data over the migration socket.
+1. Stop and delete source VM.
+1. Start the target VM using the transferred state.
+
+
+## Migration API
+
+Sending a `POST` request to `/1.0/instances/{name}` renames, moves an instance between pools, or migrates an instance to another server. In the push case, the returned operation metadata for migration is a background operation with progress data. For the pull case, it is a WebSocket operation with a number of secrets to be passed to the target server.
+
+## Live migration call stack
+
+Below is a general overview of the key functions of the live migration call stack:
+
+### [`lxd/lxd/instance_post.go`](https://github.com/canonical/lxd/blob/main/lxd/instance_post.go)
+
+1. [`instancePost`](https://github.com/canonical/lxd/blob/main/lxd/instance_post.go#L74)
+
+This function handles post requests to the `/1.0/instances` endpoint.
+
+### [`lxd/lxd/migrate_instance.go`](https://github.com/canonical/lxd/blob/main/lxd/migrate_instance.go)
+
+1. [`Do`](https://github.com/canonical/lxd/blob/main/lxd/migrate_instance.go#L87)
+
+This function performs the migration operation on the source VM for the given state and operation. It sets up the necessary WebSocket connections for control, state, and filesystem, and then initiates the migration process.
+
+### [`lxd/lxd/instance/drivers/driver_qemu.go`](https://github.com/canonical/lxd/blob/main/lxd/instance/drivers/driver_qemu.go)
+
+1. [`MigrateSend`](https://github.com/canonical/lxd/blob/main/lxd/instance/drivers/driver_qemu.go#L6436)
+
+This function controls the sending of a migration, checking for stateful support, waiting for connections, performing checks, and sending a migration offer. When performing an intra-cluster same-name migration, steps are taken to prevent corruption of volatile device configuration keys during the start and stop of the instance on both source and target.
+
+1. [`migrateSendLive`](https://github.com/canonical/lxd/blob/main/lxd/instance/drivers/driver_qemu.go#L6666)
+
+This function performs the live migration send process:
+
+1. Connect to the QEMU monitor
```

Comment:
```suggestion
1. Connect to the QEMU monitor: 
```

Reply:
Please add colons to all the initial summary phrases. Otherwise, you get this in the built doc:

<img src="https://github.com/user-attachments/assets/25082a39-1edb-4490-b301-77c93b2f963b" width="300px" />